src/CashuWallet.ts

```diff
@@ -410,47 +484,65 @@ class CashuWallet {
 	/**
 	 * Creates blinded messages for a given amount
 	 * @param amount amount to create blinded messages for
+	 * @param preference optional preference for splitting proofs into specific amounts. overrides amount param
+	 * @param count? optionally set count to derive secret deterministically. CashuWallet class must be initialized with seed phrase to take effect
 	 * @returns blinded messages, secrets, rs, and amounts
 	 */
 	private createRandomBlindedMessages(
 		amount: number,
-		amountPreference?: Array<AmountPreference>
+		amountPreference?: Array<AmountPreference>,
+		count?: number
+	): BlindedMessageData & { amounts: Array<number> } {
+		const amounts = splitAmount(amount, amountPreference);
+		return this.createBlindedMessages(amounts, count);
+	}
+
+	/**
+	 * Creates blinded messages for a according to @param amounts
+	 * @param amount array of amounts to create blinded messages for
+	 * @param count? optionally set count to derive secret deterministically. CashuWallet class must be initialized with seed phrase to take effect
+	 * @returns blinded messages, secrets, rs, and amounts
+	 */
+	private createBlindedMessages(
+		amounts: Array<number>,
+		count?: number,
+		keysetId?: string
 	): BlindedMessageData & { amounts: Array<number> } {
 		const blindedMessages: Array<SerializedBlindedMessage> = [];
 		const secrets: Array<Uint8Array> = [];
 		const rs: Array<bigint> = [];
-		const amounts = splitAmount(amount, amountPreference);
 		for (let i = 0; i < amounts.length; i++) {
-			const secret = randomBytes(32);
+			let deterministicR = undefined;
+			let secret = undefined;
+			if (this._seed && count != undefined) {
+				secret = deriveSecret(this._seed, keysetId??this.keysetId, count + i);
+				deterministicR = bytesToNumber(deriveBlindingFactor(this._seed, keysetId??this.keysetId, count + i));
+			} else {
+				secret = randomBytes(32);
+			}
```

Comment:
We should throw or terminate early if `count` is provided but `_seed` is not. This would give unexpected results otherwise. We may put error checking higher up, but putting it here is going to catch more edge cases than expecting all callers to do the validation from now into the future.

Reply:
I agree. If count is set, but not the `_seed`, we should not allow the function to run. I would say we check for that at the very top of the function, and throw an error. I think it is fair to say that if you provide a counter, you expect it to be deterministic. But now, the code will just return random, because oops the `_seed` was not set.